.github/actions/publish-nuget/New-NuGetPackage.ps1

```diff
@@ -64,6 +64,37 @@ function Get-ProjectProperty
 
 $projects = (Test-Path *.sln) ? (dotnet sln list | Select-Object -Skip 2 | Get-Item) : (Get-ChildItem *.csproj)
 
+# Download baseline version NuGet packages
+if ($EnablePackageValidation -eq 'True' -And
+    $PackageValidationBaselineVersion -And
+    !($Version -match '-(alpha|beta|preview|rc)') -And
+    $Version.Substring(0, $version.IndexOf('.')) -le $PackageValidationBaselineVersion.Substring(0, $PackageValidationBaselineVersion.IndexOf('.')))
+{
+    Write-Output 'Creating temporary project for baseline NuGet packages.'
+    dotnet new classlib -n TempProject
+    Set-Location TempProject
```

Comment:
When you enter a location and want to leave it afterwards use `Push-Location path; ...; Pop-Location` instead of `Set-Location path; ...; Set-Location ..` - it communicates the intent better.